### PR TITLE
Make CallSettings simpler to use

### PR DESCRIPTION
**Pull request by jskeet**
This should be much simpler now...

### CODE REVIEW

**Comment by jskeet** [comment]
Don't review yet - more to come...

**Comment by jskeet** [comment]
Ready for review. PTAL.

File: src/Google.Api.Gax.Grpc/CallSettingsExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        /// This may be null, in which case any call credentials in <paramref name="settings"/> are
+        /// not present in the new call settings. If both this and <paramref name="settings"/> are null,
+        /// the return value is null.</param>
+        /// <returns>A new set of call settings, or null if both parameters are null.</returns>
+        public static CallSettings WithCallCredentials(
+            this CallSettings settings,
+            CallCredentials credentials) =>
+            settings == null
+                ? CallSettings.FromCallCredentials(credentials)
+                : new CallSettings(settings.CancellationToken, credentials,
```
Comment (chrisdunelm) [inline src/Google.Api.Gax.Grpc/CallSettingsExtensions.cs]: Why not use settings.MergedWith(CallSettings.FromCallCredentials(...))?

File: src/Google.Api.Gax.Grpc/CallSettingsExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        /// This may be null, in which case any call timing in <paramref name="settings"/> are
+        /// not present in the new call settings. If both this and <paramref name="settings"/> are null,
+        /// the return value is null.</param>
+        /// <returns>A new set of call settings, or null if both parameters are null.</returns>
+        public static CallSettings WithCallTiming(
+            this CallSettings settings,
+            CallTiming timing) =>
+            settings == null
+                ? CallSettings.FromCallTiming(timing)
+                : new CallSettings(settings.CancellationToken, settings.Credentials,
```
Comment (chrisdunelm) [inline src/Google.Api.Gax.Grpc/CallSettingsExtensions.cs]: And here?

Comment (jskeet) [inline src/Google.Api.Gax.Grpc/CallSettingsExtensions.cs]: Ditto.

File: src/Google.Api.Gax.Grpc/CallSettingsExtensions.cs (hunk 1 above)
Comment (chrisdunelm) [inline src/Google.Api.Gax.Grpc/CallSettingsExtensions.cs]: Oh yes, of course, my mistake.

**Pull request merged**